test/IISIntegration.FunctionalTests/Utilities/IISDeployer.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    /// <summary>
+    /// Deployer for IIS.
+    /// </summary>
+    public partial class IISDeployer : ApplicationDeployer
+    {
+        private IISApplication _application;
+        private CancellationTokenSource _hostShutdownToken = new CancellationTokenSource();
+        private static object _syncObject = new object();
+
+        public IISDeployer(DeploymentParameters deploymentParameters, ILoggerFactory loggerFactory)
+            : base(deploymentParameters, loggerFactory)
+        {
+        }
+        public override void Dispose()
+        {
+            if (_application != null)
+            {
+                lock (_syncObject)
+                {
+                    _application.StopAndDeleteAppPool();
+                }
+
+                TriggerHostShutdown(_hostShutdownToken);
+
+                Thread.Sleep(TimeSpan.FromSeconds(3));
+            }
+
+            CleanPublishedOutput();
+            InvokeUserApplicationCleanup();
+
+            StopTimer();
+        }
+
+        public override async Task<DeploymentResult> DeployAsync()
+        {
+            using (Logger.BeginScope("Deployment"))
+            {
+                StartTimer();
+
+                var contentRoot = string.Empty;
+                _application = new IISApplication(DeploymentParameters, Logger);
+
+                // For now, only support using published output 
+                DeploymentParameters.PublishApplicationBeforeDeployment = true;
+
+                if (DeploymentParameters.PublishApplicationBeforeDeployment)
+                {
+                    DotnetPublish();
+                    contentRoot = DeploymentParameters.PublishedApplicationRootPath;
+                    //AddAccessPermissions(contentRoot, _application.AppPoolName);
+                }
+
+                var uri = TestIISUriHelper.BuildTestUri(ServerType.IIS, DeploymentParameters.ApplicationBaseUriHint);
+
+                lock (_syncObject)
+                {
+                    // To prevent modifying the IIS setup concurrently.
+                    _application.StartIIS(uri, contentRoot);
+                }
+
+                // Warm up time for IIS setup.
+                await Task.Delay(1000);
```

Comment:
👎 Find a way to check its status.

Reply:
I thought management API had a way to check status.